### PR TITLE
Updated my personal keymaps for the bfo9000 and jj50

### PR DESCRIPTION
**Pull request by AbstractKB**
## Description

Added my personal keymaps for the BFO9000 and JJ50.  This is my first PR.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* N/A

## Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/keebio/bfo9000/keymaps/abstractkb/config.h [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#ifndef CONFIG_USER_H
```
Comment (noroadsleft) [inline keyboards/keebio/bfo9000/keymaps/abstractkb/config.h]: If you could replace the include guard (the ifndef, define, and endif at the end) with just `#pragma once`?

It's simpler, and less prone to user error, so if you could make this switch?

```suggestion
#pragma once
```

File: keyboards/keebio/bfo9000/keymaps/abstractkb/config.h [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+#define CONFIG_USER_H
```
Comment (noroadsleft) [inline keyboards/keebio/bfo9000/keymaps/abstractkb/config.h]:
```suggestion
```

File: keyboards/keebio/bfo9000/keymaps/abstractkb/config.h [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+#define USE_SERIAL
+// #define USE_I2C
+
+/* Select hand configuration */
+
+#define MASTER_LEFT
+// #define MASTER_RIGHT
+// #define EE_HANDS
+
+#endif
```
Comment (noroadsleft) [inline keyboards/keebio/bfo9000/keymaps/abstractkb/config.h]:
```suggestion
```

File: keyboards/keebio/bfo9000/keymaps/abstractkb/config.h [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#ifndef CONFIG_USER_H
+#define CONFIG_USER_H
+
+#include "config_common.h"
```
Comment (noroadsleft) [inline keyboards/keebio/bfo9000/keymaps/abstractkb/config.h]: This file is included by the keyboard's `config.h` file, so Line 25 is redundant and should be deleted.

```suggestion
```

File: keyboards/keebio/bfo9000/keymaps/abstractkb/keymap.c [hunk 5]
```diff
@@ -0,0 +1,98 @@
+#include QMK_KEYBOARD_H
+
+// LAYER NAME MACROSS
+#define _BASE 0
+#define _LIST 1
+
+// Fillers to make layering more clear
```
Comment (noroadsleft) [inline keyboards/keebio/bfo9000/keymaps/abstractkb/keymap.c]: QMK defines these keycode aliases automatically. Lines 7-9 should be deleted.

https://github.com/qmk/qmk_firmware/blob/a120a000d481ed14a87988383be46c5a76604bb6/quantum/quantum_keycodes.h#L29-L31

```suggestion
```

File: keyboards/keebio/bfo9000/keymaps/abstractkb/keymap.c [hunk 6]
```diff
@@ -0,0 +1,98 @@
...
+#define _______ KC_TRNS
```
Comment (noroadsleft) [inline keyboards/keebio/bfo9000/keymaps/abstractkb/keymap.c]:
```suggestion
```

File: keyboards/keebio/bfo9000/keymaps/abstractkb/keymap.c [hunk 7]
```diff
@@ -0,0 +1,98 @@
...
+#define XXXXXXX KC_NO
```
Comment (noroadsleft) [inline keyboards/keebio/bfo9000/keymaps/abstractkb/keymap.c]:
```suggestion
```

File: keyboards/jj50/keymaps/abstractkb/keymap.c [hunk 8]
```diff
@@ -0,0 +1,139 @@
...
+You should have received a copy of the GNU General Public LicensezZZ
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#include QMK_KEYBOARD_H
+#include "action_layer.h"
+#include "rgblight.h"
+
+#define ______ KC_TRNS
+#define _DEFLT 0
```
Comment (noroadsleft) [inline keyboards/jj50/keymaps/abstractkb/keymap.c]: **This is just an FYI:**

You could replace Lines 26-29 with an enum, which would make it easier to manage adding/removing/rearranging your layers moving forward.

This is programmatically equivalent to your code:

```c
enum layers {
    _DEFLT,
    _RAISE,
    _LOWER,
    _FN
};

File: keyboards/jj50/keymaps/abstractkb/keymap.c [hunk 9]
```diff
@@ -0,0 +1,139 @@
...
+  MYRGB_TG = SAFE_RANGE
+};
+
+bool rgbinit = true;
+bool rgbon = true;
+
+const uint8_t RGBLED_RAINBOW_SWIRL_INTERVALS[] PROGMEM = {1,5,5}; //only using the first one
+
+void matrix_scan_user(void) {
+  if (rgbinit) {
```
Comment (drashna) [inline keyboards/jj50/keymaps/abstractkb/keymap.c]: This is no longer needed. 

If you use `keyboard_post_init_user` for this, it runs AFTER everything else has been initialized, so the settings will be properly retained.  :)

File: keyboards/keebio/bfo9000/keymaps/abstractkb/keymap.c [hunk 10]
```diff
@@ -0,0 +1,98 @@
...
+
+};
+
+bool rgbinit = true;
+bool rgbon = true;
+
+const uint8_t RGBLED_RAINBOW_SWIRL_INTERVALS[] PROGMEM = {1,5,5}; //only using the first one
+
+void matrix_scan_user(void) {
+  if (rgbinit) {
```
Comment (drashna) [inline keyboards/keebio/bfo9000/keymaps/abstractkb/keymap.c]: As above.  You can use `keyboard_post_init_user` for this code, instead.

**Comment by drashna** [comment]
Looks good, thanks!

**Pull request merged**